src/WebApi/Application/Repositories/IUserRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System.Threading.Tasks;
+using Domain.Entities;
+using Domain.Models;
+
+namespace Application.Repositories;
+
+public interface IUserRepository : IGenericRepository<UserModel, IEntity>
+{
+    Task<UserModel> GetUserByGoogleId(string id);
```

Comment:
Optional: provider Id would be optional in case of future extensibility. Coupling names to a specific platform or tool should be avoided if possible.

Reply:
it applies across the whole solution.